controllers/integrationpipeline/integrationpipeline_adapter.go

```diff
@@ -168,6 +169,20 @@ func GetIntegrationPipelineRunStatus(adapterClient client.Client, ctx context.Co
 		}
 	}
 
+	taskRuns, err := h.GetAllTaskRunsWithMatchingPipelineLabel(adapterClient, ctx, pipelineRun)
+	if err != nil {
+		return -1, "", err
```

Comment:
I don't know if returning `-1` in place of a blank constant is the right approach.  If you've seen it done that way elsewhere let me know, I could be wrong here.  But I would just return `intgteststat.IntegrationTestStatusTestInvalid`.  Alternatively you could add a new status for `IntegrationTestStatusUnknown` with a comment that that's to be used for edge cases like this one where we can't properly determine the status.

Reply:
That's a good point.
It's used [here](https://github.com/redhat-appstudio/integration-service/blob/a054d6cb69d24bedb7e51d23000fae8db64d2d6b/pkg/integrationteststatus/integration_test_status.go#L136) when we create a new instance of `IntegrationTestStatusDetail`type.

Currently, when we return `-1, "", err` (non-nil error) from `GetIntegrationPipelineRunStatus()` function, it gets requeued by the controller [here](https://github.com/redhat-appstudio/integration-service/blob/a054d6cb69d24bedb7e51d23000fae8db64d2d6b/controllers/integrationpipeline/integrationpipeline_adapter.go#L102) and we never get to make this constant visible to the users by calling `WriteIntegrationTestStatusesIntoSnapshot()`. This is why I went with `-1`, although I agree with you that it's not the best solution. Wdyt?

Reply:
My vote is for returning `intgteststat.IntegrationTestStatusTestInvalid` since the issue is very similar to not being able to find taskRuns referenced in childReferences.

Reply:
I've updated the code to return `intgteststat.IntegrationTestStatusTestInvalid`. Once Ryan comes back and thinks otherwise, I will make the necessary updates to the code after we collectively reach an agreement.